homework/index.js

```diff
@@ -0,0 +1,120 @@
+class App {
+  async getRepositories() {
+    if (this.repositories) {
+      return this.repositories;
+    }
+
+    const streamRepositories = await fetch('https://api.github.com/orgs/HackYourFuture/repos?per_page=100');
+    const rawRepositories = await streamRepositories.json();
+
+    const repositories = rawRepositories.map((rawRepository) => {
+      // eslint-disable-next-line no-use-before-define
+      return new Repository(rawRepository.name, rawRepository.description, rawRepository.forks, rawRepository.updated_at);
+    });
+    this.repositories = repositories;
+    return repositories;
+  }
+
+  async renderOptions(node) {
+    const select = document.createElement('select');
+
+    select.setAttribute('id', 'repo-select');
+    select.setAttribute('class', 'repo-selector');
+
+    const defaultOption = document.createElement('option');
+    defaultOption.innerText = 'Please choose a repo';
+    select.appendChild(defaultOption);
+    const repositories = await this.getRepositories();
+
+    repositories.forEach((repo) => {
+      select.appendChild(repo.renderAsOption());
+    });
+
+    node.appendChild(select);
+    node.onchange = async event => this.setRepositoryDetails(event.srcElement.value);
+  }
+
+  async setRepositoryDetails(name) {
+    try {
+      const repository = this.repositories.find(repository => repository.name === name);
+
+      document.getElementById('spinner').style.display = 'block';
```

Comment:
I'd move all these elements which you retrieve with `getElementById` to the top of your class. You only want to execute that code once; not every time you switch repo. In general, always try to think of these kinds of performance optimizations: 'general' code, which does not depend on whatever specifics you're doing inside your function, shouldn't be in that function.

Also, you're repeating this code multiple times in the scope of this one function - so more overhead (both in code clutter & in performance) eliminated by moving this somewhere on top.